### PR TITLE
Kotlin basics

### PR DESCRIPTION
**Pull request by noalecohen**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -36,4 +41,9 @@ dependencies {
...
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
-}
\ No newline at end of file
+    implementation "androidx.core:core-ktx:1.7.0"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+}
+//repositories {
+//    mavenCentral()
+//}
```
Comment (bencoz) [inline app/build.gradle]: I  think you can delete this comment

File: app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ): View? {
+        // Inflate the layout for this fragment
+        binding = FragmentAccountBinding.inflate(inflater, container, false)
+        displayTitles()
+        return binding.root
+    }
+
+    private fun displayTitles() {
+        var titles = ""
+        for(article in articleList) {
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt]: missing indentation/spaces hurts my eyes.

File: app/src/main/java/model/Article.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package model
+
+import java.io.Serializable
+
+data class Article(
+    var Title: String?,
```
Comment (bencoz) [inline app/src/main/java/model/Article.kt]: member with Capital letter?
did we lose our minds?

File: app/src/main/java/model/Article.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    var ImageUrl: String?,
```
Comment (bencoz) [inline app/src/main/java/model/Article.kt]: member with Capital letter?
did we lose our minds?

File: app/src/main/java/model/Article.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    var Author: String?,
```
Comment (bencoz) [inline app/src/main/java/model/Article.kt]: member with Capital letter?
did we lose our minds?

File: app/src/main/java/model/Article.kt [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    var Body: String?
```
Comment (bencoz) [inline app/src/main/java/model/Article.kt]: member with Capital letter?
did we lose our minds?

File: app/src/main/java/com/noalecohen/dispatcher/View Extensions.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.noalecohen.dispatcher
+
+import android.view.View
+import android.widget.Toast
+import model.BaseFragment
+
+fun BaseFragment.displayToast(str: String) = Toast.makeText(context, str, Toast.LENGTH_LONG).show()
+
+fun View.showView(boolean: Boolean): View {
+    visibility = if(boolean){
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/View Extensions.kt]: missing indentation/spaces hurts my eyes.

File: app/src/main/java/model/BaseFragment.kt [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+abstract class BaseFragment() : Fragment() {
+    lateinit var articleList: Array<Article>
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initArticleList()
+    }
+//    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+//        super.onViewCreated(view, savedInstanceState)
+//        initArticleList()
+//    }
```
Comment (bencoz) [inline app/src/main/java/model/BaseFragment.kt]: do you need the comment above?

File: app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+            if(article.Title != null) {
+                titles += article.Title
+            }
+            titles += "\n\n"
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt]: Is this the best logic to create this string?

File: app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+    private fun displayAuthor() {
+        var authors = ""
+        for(article in articleList) {
+            if(article.Author != null) {
+                authors += article.Author
+            }
+            authors += "\n\n"
+        }
+        binding.favoritesText.text = authors
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt]: same comments as in AccountFragment

File: app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        binding = FragmentHomeBinding.inflate(inflater, container, false)
+        displayBody()
+        return binding.root
+    }
+
+    private fun displayBody() {
+        var bodies = ""
+
+        for(article in articleList) {
+            val a = article.Body?.split(" ")
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt]: what is a?

File: app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+                if(a.isNotEmpty()){
+                    bodies += a[0]
+                }
+                if(a.size >= 2){
+                    bodies += " ${a[1]}"
+                }
+            }
+            bodies += "\n\n"
+        }
+        binding.homeText.text = bodies
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt]: same comments as in AccountFragment

File: app/src/main/java/com/noalecohen/dispatcher/MainActivity.kt [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+            TabLayout.OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab) {
+                when (tab.position) {
+                    0 -> fragmentManager!!.beginTransaction()
+                        .replace(binding.homePageFrameContent.id, AccountFragment()).commit()
+                    1 -> fragmentManager!!.beginTransaction()
+                        .replace(binding.homePageFrameContent.id, HomeFragment()).commit()
+                    2 -> fragmentManager!!.beginTransaction()
+                        .replace(binding.homePageFrameContent.id, FavoritesFragment()).commit()
+                    else -> {
```
Comment (bencoz) [inline app/src/main/java/com/noalecohen/dispatcher/MainActivity.kt:37]: Perhaps a log here will be better than doing nothing.

**Review by bencoz — CHANGES_REQUESTED** [review]

File: app/src/main/java/model/BaseFragment.kt (hunk 8 above)
Comment (noalecohen) [reply app/src/main/java/model/BaseFragment.kt]: I was debating which method to override.
The task required me to override the onViewCreated method, but in this way the app crashes ...
And when I override the onCreate method there was no problem.

File: app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt (hunk 11 above)
Comment (noalecohen) [reply app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt]: renamed to "articleBody"

File: app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt (hunk 9 above)
Comment (noalecohen) [reply app/src/main/java/com/noalecohen/dispatcher/AccountFragment.kt]: I guess the answer is 'no'... Can you help me please?

File: app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt (hunk 10 above)
Comment (noalecohen) [reply app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt]: I guess the answer is 'no'... Can you help me please?

File: app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt (hunk 12 above)
Comment (noalecohen) [reply app/src/main/java/com/noalecohen/dispatcher/HomeFragment.kt]: I guess the answer is 'no'... Can you help me please?

File: app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt (hunk 10 above)
Comment (BenMoveo) [reply app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt]: You guessed right. Kotlin is a high level language that allows writing code which is highly readable. Please try to think of an implementation which is more readable. A tip could be to try to create it in a descriptive way and not a creative one. Good luck.

Comment (noalecohen) [reply app/src/main/java/com/noalecohen/dispatcher/FavoritesFragment.kt]: I've tried to implement this with a listView, but the app crashed and I don't understand why...

File: app/src/main/java/com/noalecohen/dispatcher/View Extensions.kt [hunk 14]
```diff
@@ -0,0 +1,16 @@
+package com.noalecohen.dispatcher
```
Comment (BenMoveo) [inline app/src/main/java/com/noalecohen/dispatcher/View Extensions.kt]: Please don't use spaces in file names.

**Review by bencoz — APPROVED** [review]
👍

**Pull request merged**